src/LegacyPortal.js

```diff
@@ -76,11 +76,12 @@ class Portal extends React.Component {
     // Save reference for future access.
     if (overlay !== null) {
       this._mountOverlayTarget();
-      ReactDOM.unstable_renderSubtreeIntoContainer(
-        this, overlay, this._overlayTarget, (inst) => {
-          this._overlayInstance = inst;
 
-          if (this.props.onRendered) {
+      const initialRender = !this._overlayInstance;
+
+      this._overlayInstance = ReactDOM.unstable_renderSubtreeIntoContainer(
```

Comment:
this (as before) ends up being the DOM node though

Reply:
i think i moved this coe around as a hotfix for v16 but doesn't matter now oops